frontend/src/metabase/visualizations/components/settings/ChartSettingFieldsPicker.jsx

```diff
@@ -1,15 +1,31 @@
 /* eslint-disable react/prop-types */
+import { PointerSensor, useSensor } from "@dnd-kit/core";
 import cx from "classnames";
-import { Draggable, Droppable } from "react-beautiful-dnd";
+import { useCallback, useMemo } from "react";
+import { match } from "ts-pattern";
 import { t } from "ttag";
 
-import { DragDropContext } from "metabase/core/components/DragDropContext";
+import { Sortable, SortableList } from "metabase/core/components/Sortable";
 import CS from "metabase/css/core/index.css";
 import { moveElement } from "metabase/lib/arrays";
 
 import { ChartSettingFieldPicker } from "./ChartSettingFieldPicker";
 import { AddAnotherContainer } from "./ChartSettingFieldsPicker.styled";
 
+export const UNDEFINED_ITEM_KEY = "$$UNDEFINED_ITEM_KEY$$";
+export const NULL_ITEM_KEY = "$$NULL_ITEM_KEY$$";
+
+const convertField = (field) => {
+  return match(field)
+    .with(undefined, () => UNDEFINED_ITEM_KEY)
+    .with(null, () => NULL_ITEM_KEY)
+    .with(UNDEFINED_ITEM_KEY, () => undefined)
+    .with(NULL_ITEM_KEY, () => null)
+    .otherwise((x) => x);
+};
+
+const convertFieldsToSortableFields = (fields) => fields.map(convertField);
```

Comment:
That's super weird that we have different logic based on `null/undefined` values of `field` array. SortableList can't work with nullable values, so we map them to a special strings, and map back to null/undefined when we do logic in handlers